### PR TITLE
Transformation to update OBR-2 with ORC-4

### PR DESCRIPTION
**Pull request by basiliskus**
# Transformation to update OBR-2 with ORC-4

There was a missing transformation for UCSD that we're adding here: `Update OBR-2 with content from ORC-4 in the ORU result message. Replace OBR-2.1 with original content of ORC-4.1. Replace OBR-2.2 with original content of ORC-4.2`. We are expanding the `SwapPlacerOrderAndGroupNumbers` transformation to include this changes

## Issue

#1194

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java [hunk 1]
```diff
@@ -7,11 +7,16 @@
  * defines the HL7 field names.
  */
 public enum Hl7FhirMappingPath {
-    PLACER_ORDER_NUMBER(
+    PLACER_ORDER_NUMBER_ORC_2(
             "ORC.2",
             """
      Bundle.entry.resource.ofType(ServiceRequest).identifier.where(type.coding.code = 'PLAC').value
      """),
+    PLACER_ORDER_NUMBER_OBR_2(
```
Comment (jorg3lopez) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java:15]: Is this path being used?

Comment (tjohnson7021) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java:15]: I've got _no usages_ coming up too; Is this a question of whether we need this now or will need it?

File: shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy [hunk 2]
```diff
@@ -601,6 +601,39 @@ class HapiHelperTest extends Specification {
         HapiHelper.getORC21Value(sr) == null
     }
 
+
+    // OBR-2 - Placer Order Number
+    def "obr-2 methods work as expected"() {
```
Comment (jcrichlake) [inline shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy:606]: I think this test could use a better name

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java:15]: I don't think it is right now, but we're also using this FHIR paths as reference. Even if not used, it's useful for us to document all the HL7 FHIR paths

File: shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy (hunk 2 above)
Comment (basiliskus) [reply shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy:606]: This test is covering all the OBR-2 related methods in HapiHelper. Other tests in this file follow similar pattern, so if we change this one we should change the other ones too. Do you have any suggestions?

Comment (jcrichlake) [reply shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy:606]: Maybe obr-2 sets value to input or something?

Comment (basiliskus) [reply shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy:606]: It's also testing the getters. What about `obr-2 getter and setters methods work as expected`? The `work as expected` is also a pattern we've been using for naming tests

Comment (jcrichlake) [reply shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelperTest.groovy:606]: Sure, or you can leave it if you think it's fine

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java (hunk 1 above)
Comment (jorg3lopez) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/plugin/path/Hl7FhirMappingPath.java:15]: I’ll approve the PR, but I would recommend that we add paths as and when they are needed. This way, we can ensure that every documented path is relevant and necessary at the time of its inclusion.

**Review by jorg3lopez — APPROVED** [review]
Great work! The swap works as expected, good job.

**Pull request merged**